code/modules/projectiles/guns/ballistic/launchers.dm

```diff
@@ -139,19 +139,31 @@
 /obj/item/gun/ballistic/rocketlauncher/update_icon()
 	icon_state = "[initial(icon_state)]-[chambered ? "1" : "0"]"
 
-/obj/item/gun/ballistic/rocketlauncher/suicide_act(mob/user)
+/obj/item/gun/ballistic/rocketlauncher/suicide_act(mob/living/user)
+	user.visible_message("<span class='warning'>[user] aims [src] at the ground! It looks like [user.p_theyre()] about to perform a sick rocket jump!<span>", \
+		"<span class='userdanger'>You aim [src] at the ground, ready to perform a bisnasty rocket jump...</span>")
 	if(can_shoot())
-		handle_suicide(user, user, null, TRUE)
+		playsound(src, 'sound/vehicles/rocketlaunch.ogg', 80, 1, 5)
+		user.Stun(75)
```

Comment:
just curious, can you force the gun to stay in the user's hand with `put_in_active_hand(src, TRUE)`? seems a bit janky for them to drop the gun after the stun is applied.

Reply:
honestly forgot i could do that
u got it fam

Reply:
@ShizCalev It was still getting weird issues where it would drop halfway through the animation, so i just added a temporary NODROP flag instead